lib/src/value/calculation.dart

```diff
@@ -158,6 +382,200 @@ class SassCalculation extends Value {
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber || modulus is! SassNumber) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    strategyOrNumber = _simplify(strategyOrNumber);
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+
+    var args = (strategyOrNumber, numberOrStep, step);
+
+    switch (args) {
+      case (SassNumber number, null, null):
+        return SassNumber(number.value.round().toDouble())
+            .coerceToMatch(number);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassString(isVar: false) || SassNumber(),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return number_lib.roundWithStep(SassString('nearest'), number, step);
+
+      case (SassString(isVar: false), SassNumber number, SassNumber step):
+        _verifyCompatibleNumbers([number, step]);
+        return number_lib.roundWithStep(args.$1, number, step);
+
+      case (SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'), _?, _?):
+      case (SassString(isVar: true), _?, _?):
+        break;
```

Comment:
Rather than breaking and then doing additional logic after the `switch`, just return the calculation in this clause.